x-pack/plugins/lens/public/shared_components/coloring/utils.ts

```diff
@@ -83,11 +84,63 @@ export function remapStopsByNewInterval(
   return (controlStops || []).map(({ color, stop }) => {
     return {
       color,
-      stop: newMin + ((stop - oldMin) * newInterval) / oldInterval,
+      stop: newMin + ((stop - oldMin) * 100 * newInterval) / 100 / oldInterval,
     };
   });
 }
 
+// Utility to remap color stops within new domain
+export function getStopsFromColorRangesByNewInterval(
+  colorRanges: ColorRange[],
+  {
+    newInterval,
+    oldInterval,
+    newMin,
+    oldMin,
+  }: { newInterval: number; oldInterval: number; newMin: number; oldMin: number }
+) {
+  return (colorRanges || []).map(({ color, start }) => {
+    let stop = newMin + ((start - oldMin) * 100 * newInterval) / 100 / oldInterval;
```

Comment:
Is it possible to combine getStopsFromColorRangesByNewInterval and remapStopsByNewInterval, because they mostly have the same calculations?